File: src/Core/Jobs/Models/Accountant/UpcomingExpense.cs
﻿namespace Jobs.Models.Accountant;

public sealed class UpcomingExpense
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public int? CategoryId { get; set; }
    public decimal Amount { get; set; }
    public required string Currency { get; init; }
    public string? Description { get; set; }
    public DateTime Date { get; set; }
    public bool Generated { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime ModifiedDate { get; set; }
}

File: src/Accountant/accountant2/src/lib/models/server/requests/upcomingExpense.ts
export class CreateUpcomingExpense {
	constructor(
		public categoryId: number | null,
		public amount: number,
		public currency: string,
		public description: string | null,
		public date: string,
		public generated: boolean,
		public createdDate: Date | null,
		public modifiedDate: Date | null
	) {}
}

export class UpdateUpcomingExpense {
	constructor(
		public id: number,
		public categoryId: number | null,
		public amount: number,
		public currency: string,
		public description: string | null,
		public date: string,
		public generated: boolean,
		public createdDate: Date | null,
		public modifiedDate: Date | null
	) {}
}
